pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -271,8 +280,8 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 		dsw.volumesToMount[volumeName] = volumeToMount{
 			volumeName:              volumeName,
 			podsToMount:             make(map[types.UniquePodName]podToMount),
-			pluginIsAttachable:      attachable,
-			pluginIsDeviceMountable: deviceMountable,
+			pluginIsAttachable:      dsw.isAttachableVolume(volumeSpec),
+			pluginIsDeviceMountable: dsw.isDeviceMountableVolume(volumeSpec),
```

Comment:
A small thing here is isAttachableVolume() and isDeviceMountableVolume also call FindPluginBySpec which trying to search the plugin from spec. The similar process was done by dswp.desiredStateOfWorld.GetUniqueVolumeName in processPodVolumes() too. I think it would be good to optimize this part. 

I am ok if you want to use another PR to address it.

Reply:
Noted